### ISSUE TITLE
 Port fixtures from the reference parser

### ISSUE THREAD

**Issue by stasm** [projectfluent/fluent.js#239]
The reference parser has a [bunch of fixtures](https://github.com/projectfluent/fluent/tree/master/test/fixtures) in its tests. I'd like to copy them into `fluent-syntax/test/fixtures_reference` and set up a test suite to use them in order to test `fluent-syntax`'s parser.

We'll need to ignore spans and annotations for now.

**Comment by stasm** [comment]
Hey @zbraniecki, I know you said you wrote some code for this. Let me know if you'd like to take this issue. If not, I can work on it tomorrow.

**Comment by stasm** [comment]
This turns out to be much more work because of incompatibilities in the parsing behaviors of the tooling parser and the reference parser. Here's the inventory of differences I've found so far. I'm treating the behavior of the reference parser as the desired one.

- [ ] Blank lines should end `Junk`.
- [ ] Comments' `content` should include the trailing `\n`.
- [ ] Nested placeables should be possible.
- [ ] Broken `Attributes` should not break the whole message. (#237)
- [ ] Multiline `Patterns` should keep the indents of each line except the first one.
- [ ] Trailing whitespace should be kept.
- [ ] The `message[variant]` syntax should be forbidden.
- [ ] Escape sequences should be preserved in the raw value.
- [ ] `CallExpression` should allow newlines and blanklines between arguments.
- [ ] `NamedArgument`'s `val` field should be called `value`.

I've checked those which I have fixes for.

**Comment by stasm** [comment]
The discrepancies which I'd like to fix in the next version of `fluent-syntax` have been fixed on the `feature/syntax-0.6` branch. In https://github.com/projectfluent/fluent.js/commit/b32a35dfc5e4d2b3031caea9a97a67195c6e5a87 I unskipped most of the reference fixtures.

I moved #248 and #237 out to a future milestone. I'm going to file new issues for the remaining incompatibilities (currently skipped in tests). I'll use https://github.com/projectfluent/fluent.js/projects/3 to track them.

We'll close this issue when the `feature/syntax-0.6` branch is merged into `master`.
